src/lib/one_or_two/one_or_two.ml

```diff
@@ -88,6 +88,20 @@ module Monadic2 (M : Monad.S2) :
         let%map b' = f b in
         `Two (a', b')
 
+  let map_biased :
+         'a t
+      -> f:([ `One | `First | `Second ] -> 'a -> ('b, 'e) M.t)
+      -> ('b t, 'e) M.t =
+   fun t ~f ->
+    match t with
+    | `One a ->
+        M.map ~f:(fun x -> `One x) (f `One a)
+    | `Two (a, b) ->
+        let open M.Let_syntax in
```

Comment:
```suggestion
```

Reply:
The surrounded functions used the same idiom as I do, but I'll change as you requested.